sdks/java/io/kafka/build.gradle

```diff
@@ -65,26 +76,68 @@ dependencies {
   testCompile library.java.junit
   testCompile library.java.powermock
   testCompile library.java.powermock_mockito
+  testCompile library.java.testcontainers_kafka
   testRuntimeOnly library.java.slf4j_jdk14
   testRuntimeOnly project(path: ":runners:direct-java", configuration: "shadow")
-  kafkaVersion210 "org.apache.kafka:kafka-clients:2.1.0"
+  kafkaVersions.each {"kafkaVersion$it.key" "org.apache.kafka:kafka-clients:$it.value"}
```

Comment:
Will it require to include all these different Kafka client version jars into regular build?

Reply:
It should do the same what kafkaVersion210 did. I've imported KafkaIO 2.25 jar and I can't see kafka-clients 2.1.0 in the dependencies.
I'm not sure how to import a jar to Gradle locally, I will try it out on Monday.

Reply:
Could you try to build and publish KafkaIO jar with this command:
```
./gradlew -Ppublishing -PdistMgmtSnapshotsUrl=/path/to/.m2/repository/ -p sdks/java/io/kafka publishToMavenLocal
```
and check out which Kafka clients deps it comtains?

Reply:
That's super useful! Thank you!
I can't see any Kafka clients in the created jar. Here is the output of :sdks:java:io:kafka:dependencies

```
compileClasspath - Compile classpath for source set 'main'.
\--- org.apache.beam:beam-sdks-java-io-kafka:2.27.0-SNAPSHOT
     +--- org.checkerframework:checker-qual:3.7.0 FAILED
     +--- org.apache.beam:beam-vendor-guava-26_0-jre:0.1
     +--- org.apache.beam:beam-sdks-java-core:2.27.0-SNAPSHOT FAILED
     +--- org.apache.beam:beam-sdks-java-expansion-service:2.27.0-SNAPSHOT FAILED
     +--- org.slf4j:slf4j-api:1.7.30
     +--- joda-time:joda-time:2.10.5
     +--- com.fasterxml.jackson.core:jackson-annotations:2.10.2
     +--- com.fasterxml.jackson.core:jackson-databind:2.10.2
     |    +--- com.fasterxml.jackson.core:jackson-annotations:2.10.2
     |    \--- com.fasterxml.jackson.core:jackson-core:2.10.2
     +--- org.springframework:spring-expression:4.3.18.RELEASE FAILED
     \--- io.confluent:kafka-avro-serializer:5.3.2
          +--- org.apache.avro:avro:1.8.1
          |    +--- org.codehaus.jackson:jackson-core-asl:1.9.13
          |    +--- org.codehaus.jackson:jackson-mapper-asl:1.9.13
          |    |    \--- org.codehaus.jackson:jackson-core-asl:1.9.13
          |    +--- com.thoughtworks.paranamer:paranamer:2.7
          |    +--- org.xerial.snappy:snappy-java:1.1.1.3
          |    +--- org.apache.commons:commons-compress:1.8.1
          |    +--- org.tukaani:xz:1.5
          |    \--- org.slf4j:slf4j-api:1.7.7 -> 1.7.30
          +--- io.confluent:kafka-schema-registry-client:5.3.2
          |    +--- io.confluent:common-config:5.3.2
          |    |    +--- io.confluent:common-utils:5.3.2
          |    |    |    +--- org.slf4j:slf4j-api:1.7.26 -> 1.7.30
          |    |    |    \--- com.101tec:zkclient:0.10
          |    |    |         \--- org.slf4j:slf4j-api:1.6.1 -> 1.7.30
          |    |    \--- org.slf4j:slf4j-api:1.7.26 -> 1.7.30
          |    +--- org.apache.avro:avro:1.8.1 (*)
          |    +--- com.fasterxml.jackson.core:jackson-databind:2.9.10.1 -> 2.10.2 (*)
          |    \--- io.confluent:common-utils:5.3.2 (*)
          +--- io.confluent:common-config:5.3.2 (*)
          \--- io.confluent:common-utils:5.3.2 (*)

compileOnly - Compile only dependencies for source set 'main'. (n)
No dependencies

default - Configuration for default artifacts. (n)
No dependencies

implementation - Implementation only dependencies for source set 'main'. (n)
No dependencies

runtimeClasspath - Runtime classpath of source set 'main'.
\--- org.apache.beam:beam-sdks-java-io-kafka:2.27.0-SNAPSHOT
     +--- org.checkerframework:checker-qual:3.7.0 FAILED
     +--- org.apache.beam:beam-vendor-guava-26_0-jre:0.1
     +--- org.apache.beam:beam-sdks-java-core:2.27.0-SNAPSHOT FAILED
     +--- org.apache.beam:beam-sdks-java-expansion-service:2.27.0-SNAPSHOT FAILED
     +--- org.slf4j:slf4j-api:1.7.30
     +--- joda-time:joda-time:2.10.5
     +--- com.fasterxml.jackson.core:jackson-annotations:2.10.2
     +--- com.fasterxml.jackson.core:jackson-databind:2.10.2
     |    +--- com.fasterxml.jackson.core:jackson-annotations:2.10.2
     |    \--- com.fasterxml.jackson.core:jackson-core:2.10.2
     +--- org.springframework:spring-expression:4.3.18.RELEASE FAILED
     \--- io.confluent:kafka-avro-serializer:5.3.2
          +--- org.apache.avro:avro:1.8.1
          |    +--- org.codehaus.jackson:jackson-core-asl:1.9.13
          |    +--- org.codehaus.jackson:jackson-mapper-asl:1.9.13
          |    |    \--- org.codehaus.jackson:jackson-core-asl:1.9.13
          |    +--- com.thoughtworks.paranamer:paranamer:2.7
          |    +--- org.xerial.snappy:snappy-java:1.1.1.3
          |    +--- org.apache.commons:commons-compress:1.8.1
          |    +--- org.tukaani:xz:1.5
          |    \--- org.slf4j:slf4j-api:1.7.7 -> 1.7.30
          +--- io.confluent:kafka-schema-registry-client:5.3.2
          |    +--- io.confluent:common-config:5.3.2
          |    |    +--- io.confluent:common-utils:5.3.2
          |    |    |    +--- org.slf4j:slf4j-api:1.7.26 -> 1.7.30
          |    |    |    \--- com.101tec:zkclient:0.10
          |    |    |         \--- org.slf4j:slf4j-api:1.6.1 -> 1.7.30
          |    |    \--- org.slf4j:slf4j-api:1.7.26 -> 1.7.30
          |    +--- org.apache.avro:avro:1.8.1 (*)
          |    +--- com.fasterxml.jackson.core:jackson-databind:2.9.10.1 -> 2.10.2 (*)
          |    \--- io.confluent:common-utils:5.3.2 (*)
          +--- io.confluent:common-config:5.3.2 (*)
          \--- io.confluent:common-utils:5.3.2 (*)

runtimeElements - Elements of runtime for main. (n)
No dependencies

runtimeOnly - Runtime only dependencies for source set 'main'. (n)
No dependencies

testAnnotationProcessor - Annotation processors and their dependencies for source set 'test'.
No dependencies

testCompileClasspath - Compile classpath for source set 'test'.
\--- org.apache.beam:beam-sdks-java-io-kafka:2.27.0-SNAPSHOT
     +--- org.checkerframework:checker-qual:3.7.0 FAILED
     +--- org.apache.beam:beam-vendor-guava-26_0-jre:0.1
     +--- org.apache.beam:beam-sdks-java-core:2.27.0-SNAPSHOT FAILED
     +--- org.apache.beam:beam-sdks-java-expansion-service:2.27.0-SNAPSHOT FAILED
     +--- org.slf4j:slf4j-api:1.7.30
     +--- joda-time:joda-time:2.10.5
     +--- com.fasterxml.jackson.core:jackson-annotations:2.10.2
     +--- com.fasterxml.jackson.core:jackson-databind:2.10.2
     |    +--- com.fasterxml.jackson.core:jackson-annotations:2.10.2
     |    \--- com.fasterxml.jackson.core:jackson-core:2.10.2
     +--- org.springframework:spring-expression:4.3.18.RELEASE FAILED
     \--- io.confluent:kafka-avro-serializer:5.3.2
          +--- org.apache.avro:avro:1.8.1
          |    +--- org.codehaus.jackson:jackson-core-asl:1.9.13
          |    +--- org.codehaus.jackson:jackson-mapper-asl:1.9.13
          |    |    \--- org.codehaus.jackson:jackson-core-asl:1.9.13
          |    +--- com.thoughtworks.paranamer:paranamer:2.7
          |    +--- org.xerial.snappy:snappy-java:1.1.1.3
          |    +--- org.apache.commons:commons-compress:1.8.1
          |    +--- org.tukaani:xz:1.5
          |    \--- org.slf4j:slf4j-api:1.7.7 -> 1.7.30
          +--- io.confluent:kafka-schema-registry-client:5.3.2
          |    +--- io.confluent:common-config:5.3.2
          |    |    +--- io.confluent:common-utils:5.3.2
          |    |    |    +--- org.slf4j:slf4j-api:1.7.26 -> 1.7.30
          |    |    |    \--- com.101tec:zkclient:0.10
          |    |    |         \--- org.slf4j:slf4j-api:1.6.1 -> 1.7.30
          |    |    \--- org.slf4j:slf4j-api:1.7.26 -> 1.7.30
          |    +--- org.apache.avro:avro:1.8.1 (*)
          |    +--- com.fasterxml.jackson.core:jackson-databind:2.9.10.1 -> 2.10.2 (*)
          |    \--- io.confluent:common-utils:5.3.2 (*)
          +--- io.confluent:common-config:5.3.2 (*)
          \--- io.confluent:common-utils:5.3.2 (*)
```

Reply:
Sorry, I missed the request for logging the Kafka version. I'm working on it now.

Reply:
Done.